### PR TITLE
new api method: copy message

### PR DESCRIPTION
**Pull request by Flame239**
Implementation of `copyMessage` method introduced in Bot API 5.0 [(issue)](https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/issues/164)
Also in this PR I tried a new way of working with `chatId` parameter (as discussed in https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/issues/128)

### CODE REVIEW

File: telegram/src/main/kotlin/com/github/kotlintelegrambot/entities/ChatId.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.github.kotlintelegrambot.entities
+
+/**
+ * Unique identifier for the target chat or username of the target channel (in the format @channelusername)
+ */
+sealed class ChatId {
+    data class Id(val id: Long) : ChatId()
+    class Username(username: String) : ChatId() {
```
Comment (vjgarciag96) [inline telegram/src/main/kotlin/com/github/kotlintelegrambot/entities/ChatId.kt:8]: Should we call this `ChannelUsername` instead?

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/ApiClientIT.kt [hunk 2]
```diff
@@ -13,6 +14,8 @@ abstract class ApiClientIT {
 
     protected lateinit var sut: ApiClient
 
+    protected val gson = Gson()
```
Comment (vjgarciag96) [inline telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/ApiClientIT.kt:17]: I'm not really sure about this, I'd advocate for keeping this base test case as simple as possible and not start adding anything that is used by more than one test case here. 

A `Gson` instance is currently used in a couple of api client test cases only and there's not a great advantage from having it here.

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        assertEquals(username, converter?.convert(Username(username)))
+    }
+
+    @Test
+    fun `returns a converter that transforms ChatId Username class to string and adds '@'`() {
+        val converter = sut.stringConverter(ChatId::class.java, emptyArray(), retrofitMock)
+
+        val username = "battlecruiser-operational"
+        assertEquals("@$username", converter?.convert(Username(username)))
+    }
```
Comment (vjgarciag96) [inline telegram/src/test/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactoryTest.kt:51]: This looks more like a test for `ChatId` than for the `Converter`? If we weren't using retrofit library and thus not using this converters we would still want to test that the @ is added to the channel username

File: telegram/src/main/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactory.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.github.kotlintelegrambot.entities.ChatId
+import com.github.kotlintelegrambot.entities.ChatId.Id
+import com.github.kotlintelegrambot.entities.ChatId.Username
+import java.lang.reflect.Type
+import retrofit2.Converter
+import retrofit2.Retrofit
+
+class ChatIdConverterFactory : Converter.Factory() {
+    override fun stringConverter(type: Type, annotations: Array<Annotation>, retrofit: Retrofit): Converter<ChatId, String>? {
```
Comment (vjgarciag96) [inline telegram/src/main/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactory.kt:11]: Have you checked that the API operation works with both id types? I'm not 100% sure the Telegram Bot API will accept the numeric id as a string

**Review by vjgarciag96 — CHANGES_REQUESTED** [review]
Thank you so much for the contribution @Flame239! 

I left some comments, let me know if I can help with anything

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/CopyMessageIT.kt [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    private fun givenAnyCopyMessageResponse() {
+        val sendPollResponse = """
+            {"ok":true,"result":{"message_id":$ANY_RESULT_MESSAGE_ID}}
+        """.trimIndent()
+        val mockedResponse = MockResponse()
+            .setResponseCode(200)
+            .setBody(sendPollResponse)
+        mockWebServer.enqueue(mockedResponse)
+    }
```
[the remark this answers is not in the document]
Comment (Flame239) [reply telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/CopyMessageIT.kt:71]: oh that copy paste 🤦

File: telegram/src/main/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactory.kt (hunk 4 above)
Comment (Flame239) [reply telegram/src/main/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactory.kt:11]: Yes, I checked this. 
We use form urlencoded body and everything should be converted to string here (hence we override `stringConverter` method only).
If we decide to send json body where string and int representation are different, then we will need to use another deserialization approach.

**Comment by Flame239** [comment]
@vjgarciag96 fixed 👌

File: telegram/src/main/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactory.kt (hunk 4 above)
Comment (vjgarciag96) [reply telegram/src/main/kotlin/com/github/kotlintelegrambot/network/retrofit/converters/ChatIdConverterFactory.kt:11]: Oh, got it. I guess we'll also need a gson serializer/deserializer for it at some point then.

**Review by vjgarciag96 — APPROVED** [review]
LGTM! Thanks for the contribution 🙂

**Pull request merged**